gen3_tracker/meta/dataframer.py

```diff
@@ -518,111 +497,44 @@ def flattened_observations(self) -> Generator[dict, None, None]:
-                                patient[formatted_coding] = value_normalized
-                            break
-
-                for f in observation.get("focus", []):
-                    if f["reference"].startswith("Procedure/"):
-                        patient["procedure"] = f["reference"]
-                        for k, v in self.flattened_procedure(f["reference"]).items():
-                            if k not in ["id", "subject", "resourceType"]:
-                                patient[f"procedure_{k}"] = v
-
-                    elif f["reference"].startswith("Specimen/"):
-                        for k, v in self.flattened_specimen(f["reference"]).items():
-                            if k not in ["id", "subject", "resourceType"]:
-                                if k == "parent":
-                                    patient[f"specimen_{k}"] = self.get_nested_value(v, ["parent", 0, "reference"])
-                                elif k == "type":
-                                    specimen_type = self.get_nested_value(
-                                        v, ["coding", 0, "display"]
-                                    )
-                                    patient[f"specimen_{k}"] = specimen_type
-                                else:
-                                    patient[f"specimen_{k}"] = v
-
-                    elif f["reference"].startswith("Condition/"):
-                        for k, v in self.flattened_condition(f["reference"]).items():
-                            if k not in ["id", "subject", "resourceType", "encounter"]:
-                                if k == "stage":
-                                    for j, stage in enumerate(v):
-                                        stage_coding = self.get_nested_value(
-                                            stage, ["type", "coding", 0, "display"]
-                                        )
-                                        patient[f"stage_summary_{j}"] = stage_coding
-                                else:
-                                    patient[f"condition_{k}"] = v
-
-            yield patient
+        # get focus of all observations
+        for _, _, resource  in cursor.fetchall():
+            observation = json.loads(resource)
+            
+            if focus_type and 'focus' in observation and len(observation['focus']) > 0:
+                if observation['focus'][0]['reference'] != focus_type:
+                    continue
+
+            yield self.flatten_observation(observation)
+
+    def flatten_observation(self, observation: dict) -> dict:
+        # get pointer to db
+        cursor = self.connect()
+
+        # create simplified observation
+        simplified = SimplifiedResource.build(resource=observation).simplified
+
+        # extract the corresponding .focus and append its fields
+        if 'focus' in observation and len(observation['focus']) > 0:
+            # TODO: do we need to account for multiple foci?
```

Comment:
Yes.

Reply:
will throw an error so that it'll fail fast